core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/internal/CoreRemoteMongoCollectionImpl.java

```diff
@@ -0,0 +1,335 @@
+import java.util.List;
+import org.bson.BsonDocument;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+public class CoreRemoteMongoCollectionImpl<DocumentT>
+    implements CoreRemoteMongoCollection<DocumentT> {
+
+  private final MongoNamespace namespace;
+  private final Class<DocumentT> documentClass;
+  private final CoreStitchService service;
+  private final Operations<DocumentT> operations;
+
+  CoreRemoteMongoCollectionImpl(
+      final MongoNamespace namespace,
+      final Class<DocumentT> documentClass,
+      final CoreStitchService service
+  ) {
+    notNull("namespace", namespace);
+    notNull("documentClass", documentClass);
+    this.namespace = namespace;
+    this.documentClass = documentClass;
+    this.service = service;
+    this.operations = new Operations<>(namespace, documentClass, service.getCodecRegistry());
+  }
+
+  /**
+   * Gets the namespace of this collection.
+   *
+   * @return the namespace
+   */
+  public MongoNamespace getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * Get the class of documents stored in this collection.
+   *
+   * @return the class
+   */
+  public Class<DocumentT> getDocumentClass() {
+    return documentClass;
+  }
+
+  /**
+   * Get the codec registry for the CoreRemoteMongoCollection.
+   *
+   * @return the {@link org.bson.codecs.configuration.CodecRegistry}
+   */
+  public CodecRegistry getCodecRegistry() {
+    return service.getCodecRegistry();
+  }
+
+  public <NewDocumentT> CoreRemoteMongoCollection<NewDocumentT> withDocumentClass(
+      final Class<NewDocumentT> clazz
+  ) {
+    return new CoreRemoteMongoCollectionImpl<>(namespace, clazz, service);
+  }
+
+  public CoreRemoteMongoCollection<DocumentT> withCodecRegistry(final CodecRegistry codecRegistry) {
+    return new CoreRemoteMongoCollectionImpl<>(
+        namespace, documentClass, service.withCodecRegistry(codecRegistry));
+  }
+
+  /**
+   * Counts the number of documents in the collection.
+   *
+   * @return the number of documents in the collection
+   */
+  public long count() {
+    return count(new BsonDocument(), new RemoteCountOptions());
+  }
+
+  /**
+   * Counts the number of documents in the collection according to the given options.
+   *
+   * @param filter the query filter
+   * @return the number of documents in the collection
+   */
+  public long count(final Bson filter) {
+    return count(filter, new RemoteCountOptions());
+  }
+
+  /**
+   * Counts the number of documents in the collection according to the given options.
+   *
+   * @param filter  the query filter
+   * @param options the options describing the count
+   * @return the number of documents in the collection
+   */
+  public long count(final Bson filter, final RemoteCountOptions options) {
+    return operations.count(filter, options).execute(service);
+  }
+
+  /**
+   * Finds all documents in the collection.
```

Comment:
[q] Isn't there a limit to the number of documents that can be contained in a response from Stitch right? Should we be clear about that here by adding a caveat along the lines of "up to X documents"?.

Reply:
there's no limit